zephyr/Kconfig

```diff
@@ -2,10 +2,10 @@
 
 config LIBOQS
 	bool "Enable liboqs"
-	depends on ARM || ARM64 || X86 || ARCH_POSIX
+	depends on ARM || ARM64 || X86 || ARCH_POSIX || RISCV
 	help
 	  This option enables the liboqs as a Zephyr module. Currenty, the port is only
-	  available for ARM, ARM64, and x86 architectures and the native Posix simulators.
+	  available for ARM, ARM64, x86, RISCV-32, and RISCV-64 architectures and the native Posix simulators.
```

Comment:
Possibly silly question: Does "x86" also include "x86_64" ("x64")? Worthwhile adding to the documentation/help?

Reply:
i think the current implementation only supports x86 32 as the CMAKE_SIZEOF_VOID_P is being set to 4

Reply:
I just took a look at the Zephyr documentation and found x86_64 to be supported actually. Hence, setting `CMAKE_SIZEOF_VOID_P` to 4 always is incorrect (my fault). 
I will take a look at that and publish a fix once back in the office.

Reply:
OK -- withdrawing my approval then to not inadvertently merge this. Please re-request review when ready.

Reply:
should i wait until @Frauschi makes the fix?

Reply:
I just created a PR with the platform fixes for x86_64 (see #1658). Once my PR is merged, rebasing should be easy and this PR should be ready afterward.

I generally improved the 32/64 bit handling, so the detailed `if()` clauses for Riscv shouldn't be necessary anymore, resulting in all Riscv boards being supported.